spikewrap/pipeline/preprocess.py

```diff
@@ -0,0 +1,420 @@
+import json
+from typing import Dict, List, Tuple
+
+import numpy as np
+import spikeinterface.preprocessing as spre
+
+from ..configs import configs
+from ..data_classes.preprocessing import PreprocessingData
+from ..utils import utils
+from ..utils.custom_types import HandleExisting
+
+# --------------------------------------------------------------------------------------
+# Public Functions
+# --------------------------------------------------------------------------------------
+
+
+def run_preprocessing(
+    preprocess_data: PreprocessingData,
+    pp_steps: str,
+    handle_existing_data: HandleExisting,
+    log: bool = True,
+):
+    """
+    Main entry function to run preprocessing and write to file. Preprocessed
+    lazy spikeinterface recordings will be added to all sessions / runs in
+    `preprocess_data` and written to file.
+
+    Parameters
+    ----------
+
+    preprocess_data : PreprocessingData
+        A preprocessing data object that has as attributes the
+        paths to rawdata. The pp_steps attribute is set on
+        this class during execution of this function.
+
+    pp_steps: The name of valid preprocessing .yaml file (without the yaml extension).
```

Comment:
Actually, this can also take a path. In general this pattern is bad because when users pip install it will be hidden. Better to setup a config path at the start, possibly just a function `spikewrap.set_config_path()` that will store the path to use for configs in `/user/.spikewrap`.